example/async_surveyor.dart

```diff
@@ -52,10 +52,10 @@ void main(List<String> args) async {
   await driver.analyze();
 }
 
-int dirCount;
+int dirCount = 0;
 
 /// If non-zero, stops once limit is reached (for debugging).
-int _debuglimit; //500;
+int? _debuglimit; //500;
```

Comment:
There appears to be a lot of duplicated code. Consider abstracting this into a superclass or mixin.